packages/runtime/container-runtime/src/batchManager.ts

```diff
@@ -84,4 +89,21 @@ export class BatchManager {
             },
         };
     }
+
+    private checkReferenceSequenceNumber(message: BatchMessage) {
+        if (this.mc.config.getBoolean("Fluid.ContainerRuntime.DisableOpReentryCheck") === true) {
```

Comment:
Can host set feature gate in their code to ensure it's disabled, if they want to? I'm thinking of scenario where a customer (including 3rd party customer) picks up build with these changes and learns they can't move forward. I.e. it's possible we will learn there are legit cases, and we have to back off with our enforcement, but it would be nice not to have to issue patches to do so.

Reply:
I'll take your advice and make this opt-in via the runtime options with the feature gate as a break-glass solution.